lib/connectors/memory.js

```diff
@@ -679,6 +691,41 @@ Memory.prototype.updateAttributes = function updateAttributes(model, id, data, o
   }
 };
 
+Memory.prototype.replaceAttributes = function replaceAttributes(model, id, data, options, cb) {
+  if (!id) {
+    var err = new Error('You must provide an id when replacing attributes!');
+    if (cb) {
+      return cb(err);
+    } else {
+      throw err;
+    }
+  }
+  this.setIdValue(model, data, id);
+  var cachedModels = this.collection(model);
+  var modelData = cachedModels && this.collection(model)[id];
+  if (modelData) {
+    this.replace(model, data, options, cb);
+  } else {
+    cb(new Error('Could not replace attributes. Object with id ' + id + ' does not exist!'));
+  }
+};
+
+Memory.prototype.replaceOrCreate = function (model, data, options, callback) {
+  var self = this;
+  this.exists(model, self.getIdValue(model, data), options, function (err, exists) {
+    if (exists) {
+      self.replace(model, data, options, function(err, data) {
+        callback(err, data, { isNewInstance: false });
+      });
+    } else {
+      self.create(model, data, options, function (err, id) {
+        self.setIdValue(model, data, id);
```

Comment:
I guess same comment applies here, you should create a clone of `data` before modifying it.